website/docs/gator.md

```diff
@@ -0,0 +1,183 @@
+---
+id: gator
+title: The gator CLI
+---
+
+The `gator` CLI is a tool for evaluating Gatekeeper ConstraintTemplates and
+Constraints in a local environment.
+
+For now, the only subcommand is `gator test`, which allows writing unit tests
+for Constraints. We plan on adding more subcommands in the future.
+
+## Installation
+
+To install `gator`, you may either
+[download the binary](https://github.com/open-policy-agent/gatekeeper/releases)
+relevant to your system or build it directly from source.
+
+To build from source:
+```
+go get github.com/open-policy-agent/gatekeeper/cmd/gator
+```
+
+## The `gator test` subcommand
+
+### Writing Test Suites
+
+`gator test` organizes tests into three levels: Suites, Tests, and Cases:
+
+- A Suite is a file which defines Tests.
+- A Test declares a ConstraintTemplate, a Constraint, and Cases to test the
+  Constraint.
+- A Case defines an object to validate and whether the object is expected to
+  pass validation.
+
+Any file paths declared in a Suite are assumed to be relative to the Suite
+itself. Absolute paths are not allowed. Thus, it is possible to move around a
+directory containing a Suite, and the files it uses for tests.
+
+### Suites
+
+[An example Suite file](https://github.com/open-policy-agent/gatekeeper-library/blob/8765ec11c12a523688ed77485c7a458df84266d6/library/general/allowedrepos/suite.yaml).
+
+To be valid, a Suite file must declare:
+```yaml
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+```
+
+This differentiates Suites from "real" Kubernetes objects. `gator test` silently
+ignores files which do not declare these and does not attempt to run them.
+
+A Suite may declare multiple Tests, each containing different Templates and
+Constraints. Each Test in a Suite is independent.
+
+### Tests
+
+Each Suite contains a list of Tests under the `tests` field.
+
+A Test compiles a ConstraintTemplate, and instantiates a Constraint for the
+ConstraintTemplate. It is an error for the Constraint to have a different type
+than that defined in the ConstraintTemplate spec.crd.spec.names.kind, or for the
+ConstraintTemplate to not compile.
+
+### Cases
+
+Each Test contains a list of Cases under the `cases` field.
+
+A Case validates an object against a Constraint. The case may specify that the
```

Comment:
> may specify

Isn't the pass/fail required?  A test without a pass/fail requirement seems meaningless

Reply:
Oh, it defaults to "yes". Adding.